Packs/FeedAzureADConnectHealth/Integrations/AzureADConnectHealthFeed/AzureADConnectHealthFeed.py

```diff
@@ -0,0 +1,202 @@
+import demistomock as demisto
+from CommonServerPython import *
+
+from typing import Any, Callable, Dict, List, Tuple
+
+import urllib3
+from bs4 import BeautifulSoup
+import re
+
+# disable insecure warnings
+urllib3.disable_warnings()
+
+INTEGRATION_NAME = 'Microsoft Azure AD Connect Health Feed'
+
+
+class Client(BaseClient):
+    """
+    Client to use in the Microsoft Azure Feed integration. Overrides BaseClient.
+    """
+
+    def __init__(self, base_url: str, verify: bool = False, proxy: bool = False):
+        """
+        Implements class for Microsoft Azure feeds.
+        :param url: the Azure endpoint URL
+        :verify: boolean, if *false* feed HTTPS server certificate is verified. Default: *false*
+        :param proxy: boolean, if *false* feed HTTPS server certificate will not use proxies. Default: *false*
+        """
+        super().__init__(base_url, verify=verify, proxy=proxy)
+
+    def build_iterator(self) -> List:
+        """Retrieves all entries from the feed.
+        Returns:
+            A list of objects, containing the indicators.
+        """
+        result = []
+        r = self._http_request('GET', url_suffix='', full_url=self._base_url, resp_type='text')
+
+        soup = BeautifulSoup(r, 'html.parser')
+
+        def subs(text):
+            """Cleans up the parsed entries.
+            Returns:
+                A string.
+            """
+            patterns = (('comp', 'com p'), ('comm', 'com m'), ('comf', 'com f'), ('\n', ''))
+            for e in patterns:
+                text = re.sub(e[0], e[1], text)
+            return text
+
+        try:
+            pattern = re.compile("(https?:\/\/|\*\.)(\w+\.|\w+-\w+\.){1,3}\w{2,3}")
+            scraped_urls: List = [subs(pattern.match(cell.text).group(0)) for cell in soup.select(  # type: ignore # noqa
+                "tbody tr td li") if pattern.match(cell.text)]
+            for url in scraped_urls:
+                result.append({
+                    'value': url,
+                    'type': FeedIndicatorType.DomainGlob if '*' in url else FeedIndicatorType.URL,
+                    'FeedURL': self._base_url
+                })
```

Comment:
in a performance-wise perspective, instead of list-comprehension and then going over it again in a loop, have a single loop. in that case, you can also call the `re.match` once.

Reply:
I've changed this, please have a look and let me know.